src/Evolution/Systems/Cce/AnalyticSolutions/RotatingSchwarzschild.hpp

```diff
@@ -0,0 +1,144 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <complex>
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "DataStructures/SpinWeighted.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/WorldtubeData.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+class ComplexDataVector;
+/// \endcond
+
+namespace Cce::Solutions {
+
+/*!
+ * \brief Computes the analytic data for the rotating Schwarzschild solution
+ * described in \cite Barkett2019uae.
+ *
+ * \details This is a comparatively simple test which simply determines the
+ * Schwarzschild metric in transformed coordinates given by \f$\phi\rightarrow
+ * \phi + \omega u\f$, where \f$u\f$ is the retarded time.
+ */
+struct RotatingSchwarzschild : public SphericalMetricData {
+  struct ExtractionRadius {
+    using type = double;
+    static constexpr OptionString help{
+        "The extraction radius of the spherical solution"};
+    static type lower_bound() noexcept { return 0.0; }
+    static type default_value() noexcept { return 20.0; }
+  };
+  struct Mass {
+    using type = double;
+    static constexpr OptionString help{
+        "The mass of the Schwarzschild black hole"};
+    static type lower_bound() noexcept { return 0.0; }
+    static type default_value() noexcept { return 1.0; }
+  };
+  struct Frequency {
+    using type = double;
+    static constexpr OptionString help{
+        "The frequency of the coordinate rotation."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  using options = tmpl::list<ExtractionRadius, Mass, Frequency>;
+
+  static constexpr OptionString help = {
+      "Analytic solution representing a Schwarzschild black hole in a rotating "
+      "frame"};
+
+  WRAPPED_PUPable_decl_template(RotatingSchwarzschild);  // NOLINT
+
+  explicit RotatingSchwarzschild(CkMigrateMessage* /*unused*/) noexcept {}
+
+  // clang doesn't manage to use = default correctly in this case
+  // NOLINTNEXTLINE(modernize-use-equals-default)
+  RotatingSchwarzschild() noexcept {};
+
+  RotatingSchwarzschild(double extraction_radius, double mass,
+                        double frequency) noexcept;
+
+  std::unique_ptr<WorldtubeData> get_clone() const noexcept override;
+
+ protected:
```

Comment:
For my understanding, why protected and not private?

Reply:
I use the ability to inherit these functions to have a generic test tool that works for any of the spherical-coordinate based analytic solutions.